app/models/qst/server/qst_server_channel.rb

```diff
@@ -116,4 +118,16 @@ def common_to_x_attributes
     end
     attributes
   end
+
+  def infer_carrier
+    if address.present?
+      countries, carriers = Carrier.infer_from_phone_number(address.mobile_number)
+      unless carriers.empty?
+        self.carrier_guid = carriers.first.guid
```

Comment:
Why keep only the first carrier, when more than one could be returned? For messages it seems that all carriers are kept. We could consider a match when there is a non-empty intersection between the message's possible carriers and the channel's.